File: CompositePattern_Java/src/main/java/Components/MenuItem.java
package Components;

import Iterators.NullIterator;
import java.util.Iterator;

public class MenuItem extends MenuComponent
{
    public String Name;

    public String Description;

    public boolean Vegetarian;

    public double Price;

    public MenuItem(String name, String description, boolean vegetarian, double price)
    {
        Name = name;
        Description = description;
        Vegetarian = vegetarian;
        Price = price;
    }

    public boolean isVegetarian() {
        return Vegetarian;
    }

    @Override
    public void Add(MenuComponent component)
    {
        throw new UnsupportedOperationException("Can't add for menu item!");
    }

    @Override
    public void Remove(MenuComponent component)
    {
        throw new UnsupportedOperationException("Can't add for menu item!");
    }

    @Override
    public MenuComponent GetChild(int i)
    {
        throw new UnsupportedOperationException("Can't add for menu item!");
    }

    @Override
    public void Print()
    {
        System.out.println("Name: " + Name);
        System.out.println("Description: " + Description);
        System.out.println("Price: " + Price);
    }

    @Override
    public Iterator CreateIterator()
    {
        return new NullIterator();
    }
}

File: CompositePattern/CompositePattern/Components/MenuItem.cs
using CompositePattern.Iterators;

namespace CompositePattern.Components;

public class MenuItem : MenuComponent
{
    public string Name { get; set; }
    
    public string Description { get; set; }
    
    public bool Vegetarian { get; set; }
    
    public double Price { get; set; }

    public MenuItem(string name, string description, bool vegetarian, double price)
    {
        Name = name;
        Description = description;
        Vegetarian = vegetarian;
        Price = price;
    }
    
    public override void Add(MenuComponent component)
    {
        throw new InvalidOperationException("Can't add for menu item!");
    }

    public override void Remove(MenuComponent component)
    {
        throw new InvalidOperationException("Can't add for menu item!");
    }

    public override MenuComponent GetChild(int i)
    {
        throw new InvalidOperationException("Can't add for menu item!");
    }

    public override void Print()
    {
        Console.WriteLine("Name: " + Name);
        Console.WriteLine("Description: " + Description);
        Console.WriteLine("Price: " + Price);
    }

    public override IIterator CreateIterator()
    {
        return new NullIterator();
    }
}